StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift

```diff
@@ -354,6 +355,24 @@ extension LinkAccountPickerViewController: LinkAccountPickerBodyViewDelegate {
 
         let selectedPartnerAccount = selectedAccountTuple.partnerAccount
 
+        if let drawerOnSelection = selectedAccountTuple.accountPickerAccount.drawerOnSelection {
+            let genericInfoViewController = GenericInfoViewController(
+                genericInfoScreen: drawerOnSelection,
+                panePresentationStyle: .sheet,
+                didSelectPrimaryButton: { [weak self] in
+                    self?.lastDrawerOnSelectionViewController?.dismiss(animated: true)
+                },
+                didSelectURL: { _ in } // TODO(kgaidis): fix handling URL
+            )
+            genericInfoViewController.present(on: self)
+            lastDrawerOnSelectionViewController = genericInfoViewController
+
+            if !selectedAccountTuple.accountPickerAccount.allowSelection {
+                // if the account is not selectable, then we return early
+                return
+            }
```

Comment:
[Here is code logic on Stripe.js for this code and this general snippet above too](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/pull/21998/files#diff-96f25fd92e7a6ede67b5de1870616635186357cb3642022af8a41e0c25a968faR198-R216)

(I link code so reviewer can help double check, and it helps to reference later if needed)

Reply:
note that the web code has this logic:

```
      const allowSelection =
        accountDisplay?.allow_selection ?? account.allow_selection;
```

_BUT_ in iOS `accountPickerAccount` is same as `accountDisplay`....and we don't have it as a NULLABLE, so the web reference to `account.allow_selection` never can execute...we only need to keep `accountDisplay?.allow_selection` on iOS

Reply:
hmm I _generally_ (though not always) tend to use guard at the top of the function/closure, and avoid it elsewhere

it was originally added to Swift to fix the idea of "pyramid of doom" (in Swift, due to optionals, you may have to do a lot of unwrapping with if-statements, and this caused a lot of chained if-statements in sample code)

in this case, because the guard statement doesn't necessarily add such a clear value (ex. reduce pyramid of doom),  I'd tend to prefer the simpler and more common solution (if-statement)

happy to hear more convincing reasons though haha; the one reason I hesitated on the if-statement being the better solution is because this code is a bit wonky here (I copied what Stripe.js does): it early returns unless the statement passes...that can potentially give an argument that guard is more indicative of this "weirdess/early-return"; but this is getting quite subjective and in a moment of subjectivity I tend to go to the simplest possible solution/primitive